internal/db/update/update_test.go

```diff
@@ -0,0 +1,168 @@
+			upgradesInternal:      []schema.Update{},
+			upgradesExternal:      []schema.Update{dummyUpdate},
+		},
+		{
+			name:                  "Default internal schema, update external schema from v1 to v2",
+			initialSchemaInternal: []schema.Update{updateFromV0, updateFromV1},
+			initialSchemaExternal: []schema.Update{dummyUpdate},
+			upgradesInternal:      []schema.Update{},
+			upgradesExternal:      []schema.Update{dummyUpdate},
+		},
+		{
+			name:                  "Update internal schema from v1 to v2, update external schema from v1 to v2",
+			initialSchemaInternal: []schema.Update{updateFromV0, updateFromV1},
+			initialSchemaExternal: []schema.Update{dummyUpdate},
+			upgradesInternal:      []schema.Update{dummyUpdate},
+			upgradesExternal:      []schema.Update{dummyUpdate},
+		},
+		{
+			name:                  "Update internal schema from v0 to v1, external schema at v1",
+			initialSchemaInternal: []schema.Update{updateFromV0},
+			initialSchemaExternal: []schema.Update{dummyUpdate},
+			upgradesInternal:      []schema.Update{updateFromV1},
+			upgradesExternal:      []schema.Update{},
+		},
+		{
+			name:                  "Update internal schema from v0 to v2, external schema at v1",
+			initialSchemaInternal: []schema.Update{updateFromV0},
+			initialSchemaExternal: []schema.Update{dummyUpdate},
+			upgradesInternal:      []schema.Update{updateFromV1, dummyUpdate},
+			upgradesExternal:      []schema.Update{},
+		},
+		{
+			name:                  "Update internal schema from v0 to v2, update external schema from v0 to v1",
+			initialSchemaInternal: []schema.Update{updateFromV0},
+			initialSchemaExternal: []schema.Update{},
+			upgradesInternal:      []schema.Update{updateFromV1, dummyUpdate},
+			upgradesExternal:      []schema.Update{dummyUpdate},
+		},
+		{
+			name:                  "Update internal schema from v0 to v2, update external schema from v1 to v2",
+			initialSchemaInternal: []schema.Update{updateFromV0},
+			initialSchemaExternal: []schema.Update{dummyUpdate},
+			upgradesInternal:      []schema.Update{updateFromV1, dummyUpdate},
+			upgradesExternal:      []schema.Update{dummyUpdate},
+		},
+	}
+
+	for i, t := range tests {
+		s.T().Logf("%s (case %d)", t.name, i)
+
+		schema := &SchemaUpdateManager{
+			updates: map[updateType][]schema.Update{
+				updateInternal: t.initialSchemaInternal,
+				updateExternal: t.initialSchemaExternal,
+			},
+		}
+
+		db, err := NewTestDBWithSchema(schema)
+		s.NoError(err)
+
+		schema.updates[updateInternal] = append(schema.updates[updateInternal], t.upgradesInternal...)
+		schema.updates[updateExternal] = append(schema.updates[updateExternal], t.upgradesExternal...)
+
+		_, err = schema.Schema().Ensure(db)
+		s.NoError(err)
+
+		ctx := context.Background()
+		tx, err := db.BeginTx(ctx, nil)
+		s.NoError(err)
+		versions, err := query.SelectIntegers(ctx, tx, "SELECT MAX(version) FROM schemas WHERE type = 0 UNION ALL SELECT COALESCE(MAX(version), 0) FROM schemas WHERE type = 1")
```

Comment:
I think it would also be good to double check the values of `schema_internal` and `schema_external` in `internal_cluster_members`.

Reply:
Hmm, I suppose for safety I could try to spoof some values into this test too, as the upgrade test in `db_test.go` won't necessarily run `updateFromV1`.

Reply:
I've now also added a test that spoofs a pre-existing 3 member cluster from before this change, and then applies `updateFromV1` to it to validate that both the `schemas` and `internal_cluster_members` have the schema version column split in two.

Reply:
Yep mocking values in these tests was what I was expecting, thanks :)